samples/soundtest.lobster

```diff
@@ -0,0 +1,173 @@
+// Extension of shooter tutorial tut7 here with background music, enemy sounds and "hot" pause
+
+import std
+import vec
+import color
+
+fatal(gl_window("Shooter Tutorial", 640, 480))
+
+check(gl_set_font_name("data/fonts/US101/US101.ttf") and gl_set_font_size(32), "can\'t load font!")
+
+// load music and sound files
+let bgmusic = play_wav("data/sounds/Cyber-Dream-Loop.wav", -1) // by Eric Matyas from https://soundimage.org/looping-music/
+if bgmusic: sound_pause(bgmusic) else: print "can\'t load music!"
+let ufobuzz = play_wav("data/sounds/Alien_buzz2.wav", -1) // https://freesound.org/people/CommanderDerp/sounds/167921/ (Slenderman static..)
+if ufobuzz: sound_volume(ufobuzz, 0.0) else: print "can\'t load sound!"
+
+let worldsize = 20.0
+
+var playerpos = xy_0
+let playerspeed = 10
+var playerhealth = 0.0
+
+class bullet:
+    pos:xy_f
+    dir:xy_f
+
+let firerate = 0.1
+let bulletspeed = 15
+var bullets = []
+var lastbullet = gl_time()
+
+class enemy:
+    pos:xy_f
+    hp:int
+
+var enemyrate = 1.0
+let enemyspeed = 3
+let enemymaxhp = 5
+var enemies = []
+var lastenemy = gl_time()
+var closest_enemy_dist = 9999.0
+
+var score = 0
+var highscore = 0
+var playing = false
+var paused = false
+var pausetime = 0.0
+let t0 = gl_time()
+
+def renderpointytriangle(pos, dir):
+    gl_translate pos:
+        gl_rotate_z dir:
+            gl_polygon([ xy { -0.5, 0.5 }, xy_x, xy { -0.5, -0.5 } ])
+            
+def delta_time_w_pause() -> float:
+    if paused:
+        return 0.0
+    else:
+        return gl_delta_time()
+
+while gl_frame() and gl_button("escape") != 1:
+    gl_clear(color_black)
+    gl_color(color_white)
+    let t_gl = gl_time()
+    if playing:
+        // pause logic
+        if gl_button("space") == 1:
+            paused = not paused
+            if paused:
+                pausetime = t_gl
+                sound_pause(bgmusic)
+                sound_pause(ufobuzz)
+            else:
+                lastenemy  += t_gl - pausetime
+                lastbullet += t_gl - pausetime
+                sound_resume(bgmusic)
+                sound_resume(ufobuzz)
+        //
+        gl_translate float(gl_window_size()) / 2.0:
+            let scale = gl_window_size().y / worldsize
+            gl_scale(scale)
+
+            let dir = xy_f { (gl_button("d") >= 1) - (gl_button("a") >= 1),
+                          (gl_button("s") >= 1) - (gl_button("w") >= 1) }
+            let newpos = playerpos + normalize(dir) * delta_time_w_pause() * playerspeed
+            if !any(abs(newpos) > float(gl_window_size()) / scale / 2):
+                playerpos = newpos
+
+            let tomouse = normalize(gl_local_mouse_pos(0) - playerpos)
+
+            if !paused:
```

Comment:
also makes this check unnecessary, since there will be no spawning if time is not advancing.

Reply:
Yes.